pymeasure/instruments/tcpowerconversion/tccxn.py

```diff
@@ -0,0 +1,392 @@
+        bit 9: Reverse power limit,
+        bit 8: Forward power limit,
+        bit 6: MCG mode active,
+        bit 5: load power leveling active,
+        bit 4, External RF source active,
+        bit 0: RF power on.
+        """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[:2]),
+        get_process=lambda d: f"{d:016b}",
+    )
+
+    temperature = Instrument.measurement(
+        "GS\x00\x00\x00\x00",
+        """ Get heat sink temperature in deg Celsius """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[2:4]),
+        get_process=lambda d: float(d)/10,
+    )
+
+    tuner = Instrument.measurement(
+        "GS\x00\x00\x00\x00",
+        """ Get type of tuning. """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[6:]),
+        values={"none": 1, "AFT generator": 2,
+                "analog tuner": 3, "digital tuner": 4},
+        map_values=True,
+    )
+
+    power_limit = Instrument.measurement(
+        "Gp\x00\x00\x00\x00",
+        """ Get maximum power of the power supply. """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[2:4]),
+        get_process=lambda d: float(d)/10,
+    )
+
+    reverse_power_limit = Instrument.measurement(
+        "Gp\x00\x00\x00\x00",
+        """ Get maximum power reverse power. """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[18:20]),
+        get_process=lambda d: float(d)/10,
+    )
+
+    dc_voltage = Instrument.measurement(
+        "GT\x00\x00\x00\x00",
+        """ chamber DC voltage in volts. """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[6:8]),
+    )
+
+    operation_mode = Instrument.control(
+        "GS\x00\x00\x00\x00", "SO\x00%c\x00\x00",
+        """ operation mode """,
+        preprocess_reply=lambda d: struct.unpack(">H", d[4:6]),
+        values={"normal": 1, "<invalid>": 2, "pulse": 3, "ramp": 4},
+        map_values=True,
+    )
+
+    setpoint = Instrument.control(
+        "GL\x00\x00\x00\x00", "SA%c%c\x00\x00",
+        """ setpoint power level in watts """,
+        preprocess_reply=lambda d: struct.unpack(">H", d),
+        get_process=lambda d: float(d)/10,
+        set_process=int2char,
+        validator=strict_discrete_set,
+        values=range(4001),
+    )
+
+    ramp_start_power = Instrument.control(
+        "GR\x00\x00\x00\x00", "RP%c%c\x00\x00",
+        """ ramp starting power in watts""",
+        preprocess_reply=lambda d: struct.unpack(">H", d[:2]),
+        set_process=int2char,
+        validator=strict_discrete_set,
+        values=range(1, 4001),
+    )
+
+    ramp_rate = Instrument.control(
+        "GR\x00\x00\x00\x00", "RR%c%c\x00\x00",
+        """ ramp rate in watts/second""",
+        preprocess_reply=lambda d: struct.unpack(">H", d[2:]),
+        set_process=int2char,
+        validator=strict_discrete_set,
+        values=range(1, 99),
+    )
+
+    manual_mode = Instrument.control(
+        "GT\x00\x00\x00\x00", "TM\x00%c\x00\x00",
+        """ set manual tuner mode """,
```

Comment:
You can read it as well, don`t you?
```suggestion
        """Control manual tuner mode.""",
```

Reply:
I agree and implemented this below